tests/Token/FactoryTokenProviderTest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Tests\Token;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Mercure\Token\FactoryTokenProvider;
+use Symfony\Component\Mercure\Token\LcobucciFactory;
+
+final class FactoryTokenProviderTest extends TestCase
+{
+    public function testGetToken(): void
+    {
+        $factory = new LcobucciFactory('!ChangeMe!');
+        $provider = new FactoryTokenProvider($factory, ['*'], []);
+
+        self::assertStringContainsString(
+            'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJtZXJjdXJlIjp7InB1Ymxpc2giOlsiKiJdLCJzdWJzY3JpYmUiOltdfX0.',
+            $provider->getToken()
+        );
```

Comment:
not using `assertSame` as for some strange reason, PHP 7.4 + `lcobucci/jwt` result in a different signature, not sure if this is a bug, PHP 7.4/8.0 + `lcobucci/jwt:^4.0` and PHP 7.1/7.2/7.3+ `lcobucci/jwt:^3.0` all result in the same expected signature.